File: L3/java 3/Rectangle.java
public class Rectangle extends Tetragon {


    private int side1;
    private int side2;
    private int side3;
    private int side4;


    Rectangle(int side1, int side2, int side3, int side4) {
        this.side1 = side1;
        this.side2 = side2;
        this.side3 = side3;
        this.side4 = side4;
    }


    public double area() {
        if (side1 == side2) {
            return side1*side3;
        }
        return side1*side2;
    }

    public double perimeter() {
        return side1 + side2 + side3 + side4;
    }


}

File: L3/cpp 3/src/rectangle.cpp
#include <rectangle.hpp>


        Rectangle::Rectangle(int side1, int side2, int side3, int side4) {
            this->side1 = side1;
            this->side2 = side2;
            this->side3 = side3;
            this->side4 = side4;
        }

        double Rectangle::area() {
            if(side1 == side3) {
                return side1*side2;
            }
            return side1*side3;
        }

        double Rectangle::perimeter() {
            return side1 + side2 + side3 + side4;
        }
